libs/ui/src/hooks/use-activity.hook.ts

```diff
@@ -140,53 +154,51 @@ export const useAllActivity = (publicKeyHash: string, chainName: string, tokenAd
 
       if (startTime === 0) {
         setActivity(activityData);
-      } else if (activityData.length > 0) {
-        setActivity(prev => {
-          if (
-            activityData.length &&
-            prev.slice(-1)[0].data.slice(-1)[0].timestamp !== activityData.slice(-1)[0].data.slice(-1)[0].timestamp
-          ) {
-            const groupingAllDataByDates = [...prev, ...activityData].reduce(
-              (acc: Record<string, ActivityData[]>, currentItem) => {
-                if (!acc.hasOwnProperty(currentItem.title)) {
-                  return {
-                    ...acc,
-                    [currentItem.title]: currentItem.data
-                  };
-                }
-
-                return {
-                  ...acc,
-                  [currentItem.title]: [...acc[currentItem.title], ...currentItem.data]
-                };
-              },
-              {}
-            );
-
-            return Object.keys(groupingAllDataByDates).map(title => ({
-              title,
-              data: groupingAllDataByDates[title]
-            }));
+      }
+      if (
+        startTime > 0 &&
+        activityData.length > 0 &&
+        activity.length > 0 &&
+        activity.slice(-1)[0].data.slice(-1)[0].timestamp !== activityData[0].data[0].timestamp
+      ) {
+        let groupingAllDataByDates: Record<string, ActivityData[]> = {};
+
+        for (const element of [...activity, ...activityData]) {
+          if (!groupingAllDataByDates.hasOwnProperty(element.title)) {
+            groupingAllDataByDates = {
+              ...groupingAllDataByDates,
+              [element.title]: element.data
+            };
+          } else {
+            groupingAllDataByDates = {
+              ...groupingAllDataByDates,
+              [element.title]: [...groupingAllDataByDates[element.title], ...element.data]
+            };
           }
+        }
 
-          return prev;
-        });
+        const newData = Object.keys(groupingAllDataByDates).map(title => ({
+          title,
+          data: groupingAllDataByDates[title]
+        }));
+
+        setActivity(newData);
       }
 
       if (activityData.length > 0) {
         setLastTimestamp(prev => ({
           ...prev,
-          [publicKeyHash]: activityData[activityData.length - 1].data.slice(-1)[0].timestamp
+          [publicKeyHash]: activityData.slice(-1)[0].data.slice(-1)[0].timestamp
         }));
       }
     }
 
     setIsLoading(false);
   };
 
-  const fetch = debounce((startTime?: number) => {
+  const fetch = (startTime?: number) => {
```

Comment:
we use `lastTimestamp` state only once, in case `handleEndReached` func.
maybe we can somehow change logic and get rid of `lastTimestamp` state at all?

Reply:
We also need to throw through the prop with force for initial value and when we need to make additional inquiries for filters